### PR TITLE
feat(Report):  Initial Commit

### PR DESCRIPTION
**Pull request by GordonSmith**
@jbrundage - first cut at embedding the report inside the DataPatterns WUs

### CODE REVIEW

File: report/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    rowHeight: 180,
+    primaryFontSize: 20,
+    secondaryFontSize: 14,
+    primaryColor: "#494949",
+    secondaryColor: "#DDD",
+    offwhiteColor: "#FBFBFB",
+    blueColor: "#1A99D5"
+};
+
+export class Test extends Grid {
```
Comment (jbrundage) [inline report/src/index.ts]: `Test` should be `DataPatternsReport1` or something similar

File: report/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        const urlParts2 = urlParts[1].split("/report/res/index.html");
+        const wuid = urlParts2[0];
+        this._wu = Workunit.attach({ baseUrl }, wuid);
+    }
+
+    enter(domNode, element) {
+        super.enter(domNode, element);
+        domNode.style.height = (this._data.length * config.rowHeight) + "px";
+
+        this._data.forEach((row, i) => {
```
Comment (jbrundage) [inline report/src/index.ts]: It's a game-changer to be able to do this type of thing within `enter`. If certain widget classes (like this one) could be designated as having "static data" then it greatly simplifies the concept of creating widget mashups.

Comment (GordonSmith) [reply report/src/index.ts]: Whoever writes the Widget gets to dictate the shape of the data it expects (but obviously within the viz framework we want to keep it as consistent as possible)...

File: report/src/index.ts (hunk 1 above)
Comment (GordonSmith) [reply report/src/index.ts]: Go wild - this PR is a staring point for you...

**Pull request closed without merging**